File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/py_to_cpp/code_output/AreaCalculator.cpp
#include <cmath>

class AreaCalculator {
private:
    double radius;
    static constexpr double PI = 3.14159265358979323846;

public:
    AreaCalculator(double radius) : radius(radius) {}

    double calculate_circle_area() const {
        return PI * radius * radius;
    }

    double calculate_sphere_area() const {
        return 4 * PI * radius * radius;
    }

    double calculate_cylinder_area(double height) const {
        return 2 * PI * radius * (radius + height);
    }

    double calculate_sector_area(double angle) const {
        return radius * radius * angle / 2.0;
    }

    double calculate_annulus_area(double inner_radius, double outer_radius) const {
        return PI * (outer_radius * outer_radius - inner_radius * inner_radius);
    }
};

File: data/derived_cot/rq3_prompting/results/translation/r1/cpp_to_py/code_output/AreaCalculator.py
import math

class AreaCalculator:
    def __init__(self, radius: float) -> None:
        if radius < 0:
            raise ValueError("Radius cannot be negative.")
        self._radius = radius

    def calculate_circle_area(self) -> float:
        return math.pi * self._radius * self._radius

    def calculate_sphere_area(self) -> float:
        return 4 * math.pi * self._radius * self._radius

    def calculate_cylinder_area(self, height: float) -> float:
        if height < 0:
            raise ValueError("Height cannot be negative.")
        return 2 * math.pi * self._radius * (self._radius + height)

    def calculate_sector_area(self, angle: float) -> float:
        return 0.5 * self._radius * self._radius * angle

    def calculate_annulus_area(self, inner_radius: float, outer_radius: float) -> float:
        if inner_radius > outer_radius:
            raise ValueError("Inner radius cannot be greater than outer radius.")
        return math.pi * (outer_radius * outer_radius - inner_radius * inner_radius)